[MathML] Respect inline-start for mpadded lspace in RTL

Ahmad-S792 opened the pull request:
#### 3fbe9c41e3119bfccf4430dac7f3aa359ece064a
<pre>
[MathML] Respect inline-start for mpadded lspace in RTL
<a href="https://bugs.webkit.org/show_bug.cgi?id=302544">https://bugs.webkit.org/show_bug.cgi?id=302544</a>
<a href="https://rdar.apple.com/problem/164740784">rdar://problem/164740784</a>

Reviewed by Alan Baradlay.

In RenderMathMLPadded, lspace is applied as horizontal/inline-axis space
at the “start” of the content in the direction the content flows according
to the specification. Previously, lspace was always added as a positive
horizontal shift, ignoring the element&apos;s writing direction. This caused
mpadded elements inside dir=&quot;rtl&quot; to shift in the wrong direction.

This patch updates layoutBlock() to apply lspace relative to the inline
direction to fix the issue.

* Source/WebCore/rendering/mathml/RenderMathMLPadded.cpp:
(WebCore::RenderMathMLPadded::layoutBlock):
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/mpadded/mpadded-003-expected.txt: Renamed from LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/presentation-markup/mpadded/mpadded-003-expected.txt.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/mathml/presentation-markup/mpadded/mpadded-003-expected.txt: Removed.
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/mathml/presentation-markup/mpadded/mpadded-003-expected.txt: Removed.

Canonical link: <a href="https://commits.webkit.org/303070@main">https://commits.webkit.org/303070@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f4846bfc110c53679f84322d5d76a502499aadd8

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/131146 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/3473 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/42107 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/138588 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/82846 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/133016 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/3477 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/3314 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/99922 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/82846 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134092 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/2485 "Passed tests") | [⏳ 🧪 api-mac](https://ews-build.webkit.org/#/builders/API-Tests-macOS-EWS "Waiting to run tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/80617 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/8e05ebee-afb8-44d6-ba9f-40a793406d4b) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/2405 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/90 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/81834 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/110991 "Passed tests") | [⏳ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Release-WK1-Tests-EWS "Waiting to run tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/141084 "Built successfully") | | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/153/builds/3216 "Build was cancelled. Recent messages:OS: Tahoe (26.0), Xcode: 26.0.1; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (cancelled)") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/36059 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/108441 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/3266 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/2897 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/108381 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27559 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/2414 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/113728 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/56275 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/3283 "Built successfully") | [⏳ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Release-WK2-Intel-Tests-EWS "Waiting to run tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/3105 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/66679 "Build was cancelled. Recent messages:OS: Sequoia (15.5), Xcode: 16.4; Checked out pull request; ") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/3305 "Built successfully") | | | | 
| | [![loading-orange](https://github-production-user-asset-6210df.s3.amazonaws.com/3098702/291015173-08c448be-ac0a-4fd6-92a3-8165057445b7.png) 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/3213 "Build is in progress. Recent messages:OS: Tahoe (26.0), Xcode: 26.0.1; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (cancelled)") | | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay commented on Source/WebCore/rendering/mathml/RenderMathMLPadded.cpp:
```diff
@@ -119,8 +120,11 @@ void RenderMathMLPadded::layoutBlock(RelayoutChildren relayoutChildren, LayoutUn
     LayoutUnit descent = mpaddedDepth(contentDescent);
 
+    auto inlineDirection = style().writingMode().inlineDirection();
+    LayoutUnit inlineShift = (inlineDirection == FlowDirection::RightToLeft) ? -lspace() : lspace();
```
nit; I would just write it like this, creating a local variable for inlineDirection doesn’t add much value. 
`auto inlineShift = style().writingMode().inlineDirection() == FlowDirection::RightToLeft ? -lspace() : lspace();`

Ahmad-S792 replied:
Done. Thank you!

Ahmad-S792 committed:
[MathML] Respect inline-start for mpadded lspace in RTL
https://bugs.webkit.org/show_bug.cgi?id=302544
rdar://problem/164740784

Reviewed by Alan Baradlay.

In RenderMathMLPadded, lspace is applied as horizontal/inline-axis space
at the “start” of the content in the direction the content flows according
to the specification. Previously, lspace was always added as a positive
horizontal shift, ignoring the element's writing direction. This caused
mpadded elements inside dir="rtl" to shift in the wrong direction.

This patch updates layoutBlock() to apply lspace relative to the inline
direction to fix the issue.

* Source/WebCore/rendering/mathml/RenderMathMLPadded.cpp:
(WebCore::RenderMathMLPadded::layoutBlock):
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/mpadded/mpadded-003-expected.txt: Renamed from LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/presentation-markup/mpadded/mpadded-003-expected.txt.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/mathml/presentation-markup/mpadded/mpadded-003-expected.txt: Removed.
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/mathml/presentation-markup/mpadded/mpadded-003-expected.txt: Removed.

Canonical link: https://commits.webkit.org/303070@main
